File: examples/python/get_nodes.py
import requests

BASE_URL = "https://wattcoin-production-81a7.up.railway.app"

def get_nodes():
    """Fetch registered network nodes."""
    response = requests.get(f"{BASE_URL}/nodes")
    response.raise_for_status()
    return response.json()

if __name__ == "__main__":
    nodes = get_nodes()
    print(nodes)

File: examples/javascript/get_nodes.js
const BASE_URL = "https://wattcoin-production-81a7.up.railway.app";

async function getNodes() {
  try {
    const response = await fetch(`${BASE_URL}/nodes`);
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }
    const data = await response.json();
    console.log(data);
  } catch (error) {
    console.error("Error fetching nodes:", error);
  }
}

getNodes();
